python/dgl/data/reddit.py

```diff
@@ -1,65 +1,236 @@
+""" Reddit dataset for community detection """
 from __future__ import absolute_import
 
 import scipy.sparse as sp
 import numpy as np
-import os, sys
-from .utils import download, extract_archive, get_download_dir, _get_dgl_url
-from ..utils import retry_method_with_fix
+import os
+
+from .dgl_dataset import DGLBuiltinDataset
+from .utils import _get_dgl_url, generate_mask_tensor, load_graphs, save_graphs, deprecate_property
 from .. import backend as F
-from .. import convert
+from ..convert import graph as dgl_graph
+
+
+class RedditDataset(DGLBuiltinDataset):
+    r""" Reddit dataset for community detection (node classification)
+
+    .. deprecated:: 0.5.0
+        `graph` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+        `num_labels` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> num_classes = dataset.num_classes
+        `train_mask` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+            >>> train_mask = graph.ndata['train_mask']
+        `val_mask` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+            >>> val_mask = graph.ndata['val_mask']
+        `test_mask` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+            >>> test_mask = graph.ndata['test_mask']
+        `features` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+            >>> features = graph.ndata['feat']
+        `labels` is deprecated, it is replaced by:
+            >>> dataset = RedditDataset()
+            >>> graph = dataset[0]
+            >>> labels = graph.ndata['label']
+
+    This is a graph dataset from Reddit posts made in the month of September, 2014.
+    The node label in this case is the community, or “subreddit”, that a post belongs to.
+    The authors sampled 50 large communities and built a post-to-post graph, connecting
+    posts if the same user comments on both. In total this dataset contains 232,965
+    posts with an average degree of 492. We use the first 20 days for training and the
+    remaining days for testing (with 30% used for validation).
+
+    Reference: http://snap.stanford.edu/graphsage/
```

Comment:
Wrap with rst hyperlink format:

```
Reference:  `GraphSAGE<http://snap.stanford.edu/graphsage/>`_.
```

Reply:
I saw other datasets are also using plain text format. Probably leave it.

Reply:
We'll sync about the doc string format.